SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -280,4 +281,178 @@ bool SpatialCommandUtils::HasDevLoginTag(const FString& DeploymentName, bool bIs
+
+	return false;
+}
+
+bool SpatialCommandUtils::TryKillProcessWithPID(const FString& PID)
+{
+	int32 ExitCode;
+	FString StdErr;
+
+#if PLATFORM_WINDOWS
+	const FString KillCmd = TEXT("taskkill");
+	const FString KillArgs = FString::Printf(TEXT("/F /PID %s"), *PID);
+#elif PLATFORM_MAC
+	const FString KillCmd = FPaths::Combine(SpatialGDKServicesConstants::KillCmdFilePath, TEXT("kill"));
+	const FString KillArgs = FString::Printf(TEXT("%s"), *PID);
+#endif
+
+	FString KillResult;
+	bool bSuccess = FPlatformProcess::ExecProcess(*KillCmd, *KillArgs, &ExitCode, &KillResult, &StdErr);
+	bSuccess = bSuccess && ExitCode == 0;
+	if (!bSuccess)
+	{
+		UE_LOG(LogSpatialCommandUtils, Error, TEXT("Failed to kill process with PID %s. Error: %s"), *PID, *StdErr);
+	}
+
+	return bSuccess;
+}
+
+bool SpatialCommandUtils::GetProcessInfoFromPort(int32 Port, FString& OutPid, FString& OutState, FString& OutProcessName)
+{
+#if PLATFORM_WINDOWS
+	const FString Command = FString::Printf(TEXT("netstat"));
+	// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+	const FString Args = TEXT("-n -o -a");
+#elif PLATFORM_MAC
+	const FString Command = FPaths::Combine(SpatialGDKServicesConstants::LsofCmdFilePath, TEXT("lsof"));
+	// -i:Port list the processes that are running on Port
+	const FString Args = FString::Printf(TEXT("-i:%i"), Port);
+#endif
+
+	FString Result;
+	int32 ExitCode;
+	FString StdErr;
+
+	bool bSuccess = FPlatformProcess::ExecProcess(*Command, *Args, &ExitCode, &Result, &StdErr);
+
+	if (ExitCode == 0 && bSuccess)
+	{
+#if PLATFORM_WINDOWS
+		// Get the line of the netstat output that contains the port we're looking for.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(.*?:%i.)(.*)( [0-9]+)"), Port));
+#elif PLATFORM_MAC
+		// Get the line that contains the name, pid and state of the process.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(\\S+)( *\\d+).*(\\(\\S+\\))")));
+#endif
+		FRegexMatcher PidMatcher(PidMatcherPattern, Result);
+		if (PidMatcher.FindNext())
+		{
+
+#if PLATFORM_WINDOWS
+			OutState = PidMatcher.GetCaptureGroup(2 /* Get the State of the process, which is the second group. */);
+			OutPid = PidMatcher.GetCaptureGroup(3 /* Get the PID, which is the third group. */);
+			if (!GetProcessName(OutPid, OutProcessName))
+			{
+				OutProcessName = "Unknown";
+			}
+#elif PLATFORM_MAC
+			OutProcessName = PidMatcher.GetCaptureGroup(1 /* Get the Name of the process, which is the first group. */);
+			OutPid = PidMatcher.GetCaptureGroup(2 /* Get the PID, which is the second group. */);
+			OutState = PidMatcher.GetCaptureGroup(3 /* Get the State of the process, which is the third group. */);
+#endif
+			return true;
+		}
+
+#if PLATFORM_WINDOWS
+		UE_LOG(LogSpatialCommandUtils, Log, TEXT("The required port %i is not blocked!"), Port);
+		return false;
+#endif
+		 
+	}
+
+#if PLATFORM_MAC
+	if (bSuccess && StdErr.IsEmpty())
```

Comment:
it would probably be good to also check the exit code. the exit code should be 1 in this case